### DISCUSSION TITLE
Try the new Launchpad (Preview)

### DISCUSSION THREAD

**Discussion by d13** [gitkraken/vscode-gitlens#3286] [Labs]
We're excited to introduce "Launchpad," accessible now in GitLens via "Open Launchpad" command. 

The new Launchpad is categorized and gives you info at a glance on the state of your PRs, which helps you prioritize and pick the best ones to take action on, without interrupting your coding workflow. 

![Launchpad Quickpick](https://github.com/gitkraken/vscode-gitlens/assets/293282/85471fc6-1382-45bb-ba83-b01cbcef7c05)

Our team has dedicated a significant amount of effort to developing the Launchpad, and we're looking forward to rolling out new features and enhancements in future releases. 

**To try out Launchpad, run the command ‘GitLens: Open Launchpad’ from the command palette (Cmd/Ctrl+Shift+P)**

![Switch to Pre-Release](https://github.com/gitkraken/vscode-gitlens/assets/293282/f1792990-7f0d-4652-a6a8-e8a6e3474dca)

Unblock your PR reviews with Launchpad. A unified dashboard that consolidates PRs from multiple repositories, placing them front and center. Prioritize, act, and collaborate with certainty.

1. Unified View: All the crucial details are now right at your fingertips. Experience the ease of having all your pull requests displayed in a single, coherent dashboard.

2. Streamlined Workflow: Launchpad minimizes the need for context switching. You can view, manage, and interact with PRs without having to toggle between different tools or windows.

3. Immediate Actions: Launchpad provides visibility into the status and details of PRs. Letting you merge, switch to a PR's branch, leave code suggestions, and open on GitHub.

With Launchpad, you can see the most important actions needed to be taken on your Pull Requests.

![Launchpad in the Status Bar](https://github.com/gitkraken/vscode-gitlens/assets/293282/6dd6314c-ac94-41cc-9eb2-81fce503007c)

The previous version of Launchpad which was known as [Focus](https://github.com/gitkraken/vscode-gitlens/discussions/2535), is still also accessible with the "Open Launchpad in Editor" command, though our focus has been on this new experience for this release. Over the next few releases, we will be bringing more features and enhancement to the “in-editor” version of Launchpad.

The evolution of this tool is just beginning, and we can't wait to share the enhanced capabilities that will further streamline your workflow in future updates. Stay tuned!

Try it out and let us know what you think below!

**Comment by out-of-phaze** [comment]
The launchpad is useful, but the lack of customizability for categories is disappointing; I was hoping for something like the categories for issues and pull requests in the extension Github Pull Requests, which have editable queries. My organization's workflow has a number of additional restrictions for mergeability (we require 24h since approval with one reviewer, or immediately with two reviewers) that aren't reflected by Github's own merge requirement settings. Having more flexibility outside of just the native Github restrictions would make this feature much more useful for me, personally.

**Comment by asportnoy** [comment]
One issue is that it shows pull requests as mergable even if it's not pointing to the base branch. In my case, these PRs are not actually ready to be merged as they are waiting for the base branch to be merged first, so I don't want to see them marked as mergable in the launchpad until the base branch is merged. An option to control this would be nice.

Also fyi the feedback button is broken. Should be 3286 not 3268.

**Reply by d13** [reply]
Thanks for the feedback! The link will be fixed with tomorrow's pre-release.

**Comment by Bastian0930** [comment]
One thing I'd like to request is the ability to login to my pro account (which is linked to my real name) but use any account as the one that uses launch pad. I don't want to make pull requests for hobby projects on my more serious account - it would be extremely helpful if this was a option!

**Reply by madoke** [reply]
I second this one. In my case I use the same github account for personal and professional contexts under different organizations. I can see that there's already an ignored repositories setting, so perhaps we could use that to filter repos by github organization

**Comment by nihonjinrxs** [comment]
I would also like to be able to specify an org (or in general, a filter on repos) for repos to include. At the moment, the only customization I've found in the settings for this is repos to exclude.

I have different VSCode installations on different machines that are targeted for specific purposes. On a machine used for work for a specific org, I'd like to only see things relevant to that org, or to repos I specify that I need to work with for that work, uncluttered by any other work on other orgs or repos.

**Reply by eamodio** [reply]
In the latest pre-release you can set `gitlens.launchpad.ignoredOrganizations` to exclude specific organizations, but we don't yet have a allow-list.

We are also thinking about adding a "scope" option to the Launchpad that will be something like `everything` (which is what we have today), `openOrgs` (find the unique set of orgs in the open repos and include only those), and `openRepos` (find the set of opened repos and include only those).

Let us know what you think.

**Comment by Bruncky** [comment]
Launchpad is awesome, I'm loving it!

The only request for me would be to enable us to log into a GitHub Enterprise account, just like we can for the GitKraken account. Right now, there's not much use for me to see open PRs for my personal account since I'm always working on the company's account and generating code/PRs there.

Cheers!
